src/index.ts

```diff
@@ -201,18 +249,38 @@ class IconFontBuildr {
   async getIcons () {
 
     const filePaths = ( await globby ( '*.svg', { cwd: this.paths.cache.icons, absolute: true } ) ).sort (), // Ensuring the order is fixed
-          codepointStart = '\uE000', // Beginning of Unicode's private use area
+          codepointStart = '\uE000'.charCodeAt ( 0 ), // Beginning of Unicode's private use area
+          usedCodepoints: Map<String, String> = new Map(),
           icons = {};
 
-    filePaths.forEach ( ( filePath, index ) => {
+    let codepointOffset = 0;
+
+    filePaths.forEach ( ( filePath ) => {
+      const basename = path.basename ( filePath, path.extname ( filePath ) ),
+            icon = this.config.icons [ basename ];
+
+      let codepoints = icon.codepoints;
+
+      if ( codepoints && codepoints.length ) {
+
+        codepoints.forEach( codepoint => {
+          if ( usedCodepoints.has( codepoint ) ) exit ( `Codepoint ${ codepoint } (${ codepoint.charCodeAt ( 0 ).toString ( 16 ) }) of glyph "${ icon.name }" is colliding with glyph "${ usedCodepoints.get ( codepoint ) }".` );
```

Comment:
I would prefer this potentially critical (in the sense that the app might exit) check to be performed near the others.

Reply:
This is influenced by the dynamic _legacy_ codepoints that just count up from `\uE000`.

If you want to perform this check earlier in the code, you'd have to assign the variable codepoints sooner too.

Reply:
> If you want to perform this check earlier in the code, you'd have to assign the variable codepoints sooner too

If we have to assign a codepoint (rather than have it provided via the config) I don't think it can ever clash with anything, as in that case we will just try assigning it the next one and so on.

Reply:
If you assign a glyph a dynamic codepoint (e.g. `\uE042`) and then later encounter a glyph whose config has the same codepoint defined, you run into this exact check.

One way to solve this would be to just assign the next codepoint to the dynamically defined glyph, but that would yield different codepoints than before for legacy configurations if they added glyphs using the advanced configuration that would cause this exact problem and I didn't wanna go there 😀

Reply:
I think we are now preforming the proper checks for this use case, is this not the case?